src/main/java/com/gtnewhorizon/structurelib/net/UpdateDebugWriterModePacket.java

```diff
@@ -0,0 +1,51 @@
+package com.gtnewhorizon.structurelib.net;
+
+import com.gtnewhorizon.structurelib.item.ItemDebugStructureWriter;
+import cpw.mods.fml.common.network.simpleimpl.IMessage;
+import cpw.mods.fml.common.network.simpleimpl.IMessageHandler;
+import cpw.mods.fml.common.network.simpleimpl.MessageContext;
+import io.netty.buffer.ByteBuf;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+
+import java.io.IOException;
+
+public class UpdateDebugWriterModePacket implements IMessage, IMessageHandler<UpdateDebugWriterModePacket, IMessage> {
+    NBTTagCompound tagCompound;
+
+    public UpdateDebugWriterModePacket() {}
+
+    public UpdateDebugWriterModePacket(ItemStack itemStack) {
+        this.tagCompound = itemStack.copy().getTagCompound();
+    }
+
+    @Override
+    public void fromBytes(ByteBuf buf) {
+        PacketBuffer pb = new PacketBuffer(buf);
```

Comment:
Please use `cpw.mods.fml.common.network.ByteBufUtils#readTag` instead. Even if it is effectively the same code, reusing the forge function would give any future reader a confidence that this *is* the right way to write these NBT tags.